src/main/java/io/supertokens/inmemorydb/Start.java

```diff
@@ -1246,4 +1251,144 @@ public int deleteUserMetadata(String userId) throws StorageQueryException {
+            }
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public String[] getRolesForUser(String userId) throws StorageQueryException {
+        try {
+            return UserRoleQueries.getRolesForUser(this, userId);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public String[] getUsersForRole(String role) throws StorageQueryException, UnknownRoleException {
+        try {
+            if (!this.doesRoleExist(role)) {
+                throw new UnknownRoleException();
+            }
+            return UserRoleQueries.getUsersForRole(this, role);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+
+    }
+
+    @Override
+    public String[] getPermissionsForRole(String role) throws StorageQueryException, UnknownRoleException {
+        try {
+            if (!this.doesRoleExist(role)) {
+                throw new UnknownRoleException();
+            }
+            return UserRoleQueries.getPermissionsForRole(this, role);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public String[] getRolesThatHavePermission(String permission) throws StorageQueryException {
+        try {
+            return UserRoleQueries.getRolesThatHavePermission(this, permission);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public int deleteRole(String role) throws StorageQueryException {
+        try {
+            return UserRoleQueries.deleteRole(this, role);
+        } catch (StorageTransactionLogicException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public String[] getRoles() throws StorageQueryException {
+        try {
+            return UserRoleQueries.getRoles(this);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public boolean doesRoleExist(String role) throws StorageQueryException {
+        try {
+            return UserRoleQueries.doesRoleExist(this, role);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public int deleteAllRolesForUser(String userId) throws StorageQueryException {
+        try {
+            return UserRoleQueries.deleteAllRolesForUser(this, userId);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public boolean deleteRoleForUser_Transaction(TransactionConnection con, String userId, String role)
+            throws StorageQueryException {
+        Connection sqlCon = (Connection) con.getConnection();
+        try {
+            return UserRoleQueries.deleteRoleForUser_Transaction(this, sqlCon, role, userId);
```

Comment:
Why does this take role, userId whereas the plugin interface function takes userId, role?